src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlException.cs

```diff
@@ -99,6 +99,207 @@ public override void GetObjectData(SerializationInfo si, StreamingContext contex
         /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlException.xml' path='docs/members[@name="SqlException"]/Source/*' />
         override public string Source => TdsEnums.SQL_PROVIDER_NAME;
 
+#if NET6_0_OR_GREATER
+        /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlException.xml' path='docs/members[@name="IsTransient"]/Source/*' />
+        public override bool IsTransient
+        {
+            get
+            {
+                if (Errors.Count == 0)
+                    return false;
+
+                foreach (SqlError err in Errors)
```

Comment:
Instead of this, which is a new list, would recommend using existing logic to determine whether the exception is going to be retried on.